File: 0137-single-number-ii/0137-single-number-ii.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        numsMap={}
        for num in nums:
            print(num)
            if num in numsMap:
                numsMap[num] +=1
            else:
                numsMap[num]=1
        for key,val in numsMap.items():
            if val==1:
                return key
        return -1
        

File: 0137-single-number-ii/0137-single-number-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let newMap={}
for (let i=0;i<=nums.length;i++){
    if(nums[i] in newMap){
        newMap[nums[i]] +=1
    }else{
        newMap[nums[i]] =1
        
    }
}
for (const [key,value] of Object.entries(newMap)){
    if (value ===1){
        return key
    }

}
return -1
};